### DISCUSSION TITLE
Access DiscountNode in cart checkout validation

### DISCUSSION THREAD

**Discussion by jacknavynicy** [Shopify/function-examples#445] [Feedback: Cart and Checkout Validation API]
I am developing an app that created the Product Discounts' function

And due to the requirements, I need to exclude some products from the discount settings to be checkout from the cart, so I am created on the Cart and Checkout Validation Function for this purpose

Usually if in the Product Discounts' I can in the query the discountNode by the following
`
  discountNode {
    metafield(namespace: "NAME", key: "discount-config") {
      value
    }
  }
`
 
So I wanted to get the same thing in the Cart and Checkout Validation Function, but I received the error below
![image](https://github.com/Shopify/function-examples/assets/133320093/a34a9cc9-a406-4ab3-838e-67768bc53e1f)

  Is the Cart and Checkout Validation Function can't access the discountNode from other discount function?
  
 We prefer if we can access the info like this
 Or is there any alternative way I can do this? is the Cart and Checkout Validation Function allowed custom inputs from customer? (eg: product_Ids)

**Comment by thosakwe** [comment]
Maybe you can copy the list of excluded products into the cart and checkout validation function configuration metafield.

**Comment by nickwesselman** [comment]
Reference for the cart and checkout input can be found here:
https://shopify.dev/docs/api/functions/reference/cart-checkout-validation/graphql/input

I think what you are really asking for is to have information on the applied discounts in the input, along with their metafields?

Another option would be to use shop metafields, which both functions have access to.

**Reply by jacknavynicy** [reply]
Currently the discount info(eg: which products can be treated as free gift), is stored in the Discount Node Metafields

What we hope is if we can access the Discount Node from the Cart and Checkout Validation Function to get those products, so we can add some validation on these products

Your suggestion of using shop metafields can be considering as the alternative way to let user to set the products, thanks for your suggestion!

**Reply by jacknavynicy** [reply]
Hi @nickwesselman I think the alternative way is not suitable for our case, due to we need the current active's discount

ya as you mentioned earlier, we hope we can get the current active discount's metafields so we can have some validation on the products

**Comment by Waynenabors** [comment]
Our use case if very similar. It's possible to create multiple function powered discounts but there is no way to find out what actually ended up applying across all of them. So when a merchant wants to set a discount limit across ALL discounts they can't.

I think Discounts Allocator would be able to do this but that is not an option for public apps.
